### PR TITLE
No underscore in connect-mixin API

### PR DESCRIPTION
**Pull request by keanulee**

### CODE REVIEW

**Comment by keanulee** [comment]
`0.9.0-pre.1` published to `@next` tag on NPM

**Comment by keanulee** [comment]
@eterna2 I couldn't figure out how to type an abstract constructor in TS (saw [this](https://stackoverflow.com/questions/36886082/abstract-constructor-type-in-typescript) but didn't work). Because of this, we're making [`LitElement` itself not abstract anymore](https://github.com/Polymer/lit-element/pull/228). I also couldn't return an abstract class in the mixin function.

We played with the idea of `mapState/DispatchToProps`, but ultimately abandoned it because it didn't play well with property typings (`return { prop: 'foo' }` wouldn't check that `prop` was on the element) and thought it was straightforward enough to just assign to `this.prop` in `stateChanged`. The implementation is simple though (`Object.assign(this, maStateToProps(state))`), so you could adopt this style if it better meets your needs.

File: README.md [hunk 1]
```diff
@@ -9,49 +9,94 @@ These are vanilla JavaScript methods that can be used regardless of which
 frameworks or libraries your application is written in.
```
Comment (frankiefu) [inline README.md]: We should probably add build status and npm badges, e.g.
```
[![Build Status](https://travis-ci.org/Polymer/pwa-helpers.svg?branch=master)](https://travis-ci.org/Polymer/pwa-helpers)
[![Published on npm](https://img.shields.io/npm/v/pwa-helpers.svg)](https://www.npmjs.com/package/pwa-helpers)
```

**Comment by eterna2** [comment]
@keanulee

To also accept abstract class, this is what I did.

`ConstructorType|HTMLElement`

But yeah, I can't return an abstract class either. But at least it let me pass in an abstract class. Might need to add an `implements` myself to mimic the abstract part.

**Comment by keanulee** [comment]
I'm confused - what exactly is the signature? It's currently:

```ts
type Constructor<T> = new(...args: any[]) => T;

export const connect =
  <S>(store: Store<S>) =>
  <T extends Constructor<HTMLElement>>(baseElement: T) =>
```

Since we can't return an abstract class anyway, the mixin would need to implement any abstract methods, and we wouldn't be able to do this for any arbitrary subclass of `HTMLElement` since we wouldn't know what's required. Not going to block this PR on this feature since we already don't allow abstract classes.

**Comment by keanulee** [comment]
@frankiefu PTAL and try out 0.9.0-pre.4. https://github.com/Polymer/pwa-starter-kit/pull/240 seems to  be good.

**Pull request merged**